iml-system-test-utils/src/docker.rs

```diff
@@ -0,0 +1,223 @@
+use crate::{iml, CheckedStatus, iml::IML_DOCKER_PATH};
+use iml_systemd;
+use iml_wire_types::Branding;
+use std::{io, thread, time, str};
+use tokio::{
+    fs::{canonicalize, File},
+    io::AsyncWriteExt,
+    process::Command,
+};
+
+pub struct DockerSetup {
+  pub use_stratagem: bool,
+  pub branding: Branding,
+}
+
+pub async fn docker() -> Result<Command, io::Error> {
+    let mut x = Command::new("docker");
+
+    let path = canonicalize(IML_DOCKER_PATH).await?;
+
+    x.current_dir(path);
+
+    Ok(x)
+}
+
+pub async fn deploy_iml_stack() -> Result<(), io::Error> {
+    iml_systemd::start_unit_and_wait("iml-docker.service".into(), 400).await?;
+
+    Ok(())
+}
+
+pub async fn get_docker_service_count() -> Result<u16, io::Error> {
+  let mut x = docker().await?;
+
+  let services = x.arg("service").arg("ls").output().await?;
+
+  let output = str::from_utf8(&services.stdout).expect("Couldn't read docker service list.");
+  let cnt: u16 = output.lines().count() as u16 - 1; // subtract the header
+
+  Ok(cnt)
+} 
+
+pub async fn remove_iml_stack() -> Result<(), io::Error> {
+    iml_systemd::stop_unit("iml-docker.service".into()).await?;
+
+    let mut x = docker().await?;
+
+    x.arg("stack").arg("rm").arg("iml").checked_status().await?;
+
+    // Make sure there are no services.
+    let one_sec = time::Duration::from_millis(1000);
+    let mut count = get_docker_service_count().await?;
+    while count > 0 {
+      println!("Waiting on all docker services to be shut down.");
+      thread::sleep(one_sec);
+      count = get_docker_service_count().await?;
+    }
+    
+    Ok(())
+}
+
+pub async fn system_prune() -> Result<(), io::Error> {
+    let mut x = docker().await?;
+
+    x.arg("system")
+        .arg("prune")
+        .arg("--force")
+        .arg("--all")
+        .checked_status()
+        .await?;
+
+    Ok(())
+}
+
+pub async fn volume_prune() -> Result<(), io::Error> {
+    let mut x = docker().await?;
+
+    x.arg("volume")
+        .arg("prune")
+        .arg("--force")
+        .checked_status()
+        .await?;
+
+    Ok(())
+}
+
+pub async fn iml_stack_loaded() -> Result<bool, io::Error> {
+    let x = iml::list_servers().await?.status().await?;
+
+    Ok(x.success())
+}
+
+pub async fn wait_for_iml_stack() -> Result<(), io::Error> {
```

Comment:
Why do we need this fn? the systemd service should wait until services are loaded and the api is accessible:

https://github.com/whamcloud/integrated-manager-for-lustre/blob/fc8a24bf662790622dd13ead7c2279dd88fcd257/iml-docker.service#L14

Reply:
Yeah I don't think we need this function. I'll remove.